docs/en/ingest-management/elastic-agent/configuration/providers/kubernetes-provider.asciidoc

```diff
@@ -0,0 +1,429 @@
+[[kubernetes-provider]]
+= Kubernetes Provider
+
+Provides inventory information from Kubernetes.
+
+
+[discrete]
+== Provider configuration
+
+[source,yaml]
+----
+providers.kubernetes:
+  node: ${NODE_NAME}
+  scope: node
+  #kube_config: /Users/elastic-agent/.kube/config
+  #sync_period: 600s
+  #cleanup_timeout: 60s
+  resources:
+    pod:
+      enabled: true
+----
+
+`node`:: (Optional) Specify the node to scope {agent} to in case it
+cannot be accurately detected when running {agent} in host network
+mode.
+`cleanup_timeout`:: (Optional) Specify the time of inactivity before stopping the
+running configuration for a container. This is `60s` by default.
+`sync_period`:: (Optional) Specify the timeout for listing historical resources.
+`kube_config`:: (Optional) Use the given config file as configuration for Kubernetes
+client. If kube_config is not set, the KUBECONFIG environment variable will be
+checked and will fall back to InCluster if not present.
```

Comment:
I am not sure how helpful the `will fall back to InCluster if not present` is to a user. Maybe we could state exactly the steps that are followed